tutorials/large-messages/index.mdx

```diff
@@ -0,0 +1,377 @@
+    An `scaleway_object_bucket_acl` is added on the bucket to make it private and output the bucket name to use it in your producer.
+
+5. Next, add these ressources to create a NATS account an your NATS credential file:
+    ```terraform
+    resource "scaleway_mnq_nats_account" "large_messages" {
+      name = "nats-acc-large-messages"
+    }
+
+    resource "scaleway_mnq_nats_credentials" "large_messages" {
+      name       = "nats-large-messages-creds"
+      account_id = scaleway_mnq_nats_account.large_messages.id
+    }
+
+    resource "local_file" "nats_credential" {
+      content         = scaleway_mnq_nats_credentials.large_messages.file
+      filename        = "large-messages.creds"
+      file_permission = 644
+    }
+
+    output "nats_url" {
+      value       = scaleway_mnq_nats_account.large_messages.endpoint
+      description = "NATS url to use with the producer script"
+    }
+    ```
+    We also output the NATS server url.
+
+6. You can run `terraform init` and `terraform apply` to create the ressources in your Scaleway account. 
+
+## Create the producer
+
+As said before, the producer will be a simple shell script.
+
+1. Create a file `upload_img.sh` 
+    In it create two variables that use terraform command to get bucket name and NATS Url:
+    ```bash
+    SCW_BUCKET="$(terraform output bucket_name)"
+    SCW_NATS_URL="$(terraform output nats_url)"
+    ```
+
+2. Next configure the NATS cli with the NATS url and the credentials file.
+    ```bash
+    nats context save large-messages --server=$SCW_NATS_URL --creds=./large-messages.creds
+    nats context select large-messages
+    ```
+    Our script take the file path that we want to upload as first parameter.
+
+3. To upload our file we gonna use the AWS cli configured with Scaleway endpoint and credentials because the Scaleway Object storage is fully compliant with S3.
+    Pass the path to the AWS cli command like that:
+    ```bash
+    aws s3 cp $1 s3://$SCW_BUCKET
+    ```
+
+4. Past this command to make the script send a message to your NATS server with the name of the uploaded file in it.
+    ```bash
+    nats pub large-messages $(basename $1)
+    ```
+
+## Create the consumer
+
+For the consumer, we will continue to play with the Scaleway ecosystem and we gonna use a Serverless Function in Python.
+1. create a folder `function`, in it create an other folder `handler` with a file named `large_messages.py`.
+2. Aside from the `handler` folder, create a `requirements.txt` file.
+    You should have a structure like that:
+    ```
+    function/
+    ├── handler/
+    │   └── large_messages.py
+    └── requirements.txt
+    ```
+
+3. In the serverless function we will use two python libraries, `boto3` also configured like AWS cli to download and upload files in the bucket and `img2pdf` to convert our image in pdf.
```

Comment:
```suggestion
3. Within the serverless function, we use two Python libraries. First, `boto3` is configured similarly to AWS CLI, enabling the download and upload of files to the bucket. Second, `img2pdf` is used to convert our image into a PDF format.
```

Reply:
Since this isn't an instruction but an explanation, I'd put it "outside" the numbered step, and make the "In the `requirements.txt` add:" part the numbered step.

Reply:
Done, i've put the explanation after the requirement.txt content